### ISSUE TITLE
dotnet not found in VSCode on clean install on MacOSX

### ISSUE THREAD

**Issue by johanhaleby** [ionide/ionide-vscode-fsharp#1271]
I originally posted this issue on [stackoverflow](https://stackoverflow.com/questions/59336094/getting-f-working-in-vscode-when-dotnet-is-not-found-on-macosx) so my issue is copied from there.

I'm completely new to dotnet and vscode so I might be missing something trivial. I just want to get F# running in VSCode on my Mac. I first tried installing dotnet from brew:

```
brew cask install dotnet-sdk
```

This seems to be working fine from the command line since I could do both `dotnet new` and `dotnet build`:

```
dotnet new console -lang F# -o myproj
```

Now I installed Visual Studio Code, also from brew:

```
brew cask install visual-studio-code
```

I then installed the [Ionide-fsharp](https://marketplace.visualstudio.com/items?itemName=Ionide.Ionide-fsharp) plugin, opened the `myproj` folder from vscode and restarted it. I'm then greeted with:

[![error][1]][1]

I then searched for `FSharp.dotnetRoot` in the settings and tried to configure this:

[![enter image description here][2]][2]

But it didn't help (I also tried setting the folder to `/usr/local/share/dotnet` and `/usr/local/share/dotnet/sdk/3.1.100` but nothing worked).

So I completely uninstalled everything (including wiping my `~/.vscode` folder) from brew and re-installed everything from the official [Microsoft documentation](https://docs.microsoft.com/en-us/dotnet/fsharp/get-started/install-fsharp). In the docs it says that the F# support should be built-in to vscode on macosx, but it still recommended me to install Ionide-fsharp when I opened the folder again, so I obeyed. But I just ran into exactly the same problem again.

So my question is: How can I install vscode with working F# support on MacOSX?

Here's my output from vscode "about":

```
Version: 1.41.0
Commit: 9579eda04fdb3a9bba2750f15193e5fafe16b959
Date: 2019-12-11T17:58:38.338Z
Electron: 6.1.5
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Darwin x64 19.2.0
```

and this is from `dotnet --info`:

```
.NET Core SDK (reflecting any global.json):
 Version:   3.1.100
 Commit:    cd82f021f4

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.15
 OS Platform: Darwin
 RID:         osx.10.15-x64
 Base Path:   /usr/local/share/dotnet/sdk/3.1.100/

Host (useful for support):
  Version: 3.1.0
  Commit:  65f04fb6db

.NET Core SDKs installed:
  3.1.100 [/usr/local/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

  [1]: https://i.stack.imgur.com/LFyoX.png
  [2]: https://i.stack.imgur.com/HIXr3.png

**Comment by Krzysztof-Cieslak** [comment]
CC: @baronfel any idea?

**Comment by baronfel** [comment]
Next steps would be to check the output logs for details about what the SDK root is, and the ionide diagnostics info might help as well.

**Comment by ValeTheVioletMote** [comment]
I'm having the same issue on Windows, fresh install of .NET Core. dotnet not found, despite being able to run `dotnet` and `dotnet fsi` rather easily from command prompt...

**Comment by johanhaleby** [comment]
@ValeTheVioletMote Did you manage to get any progress? I wonder how everyone else seem to get things working?

**Comment by martinbryant** [comment]
Same issue for me too again on a fresh install of macOs. 

Here's my output from vscode "about":

> Version: 1.42.1
Commit: c47d83b293181d9be64f27ff093689e8e7aed054
Date: 2020-02-11T14:44:27.652Z
Electron: 6.1.6
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Darwin x64 18.7.0

and this is from dotnet --info:

> Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.14
 OS Platform: Darwin
 RID:         osx.10.14-x64
 Base Path:   /usr/local/share/dotnet/sdk/3.1.102/
Host (useful for support):
  Version: 3.1.2
  Commit:  916b5cba26
.NET Core SDKs installed:
  3.1.102 [/usr/local/share/dotnet/sdk]
.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.2 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download

I re-installed my whole os to try and make this work and to be honest it is a bit frustrating. Have not got this issue on my work Windows 10 machine.

**Comment by ntwilson** [comment]
I experience this when I try to use `netcore` as the FSAC runtime on my Windows 10 machine.  Interestingly, even with logging set to verbose, I see _no_ F# logs whatsoever in my Output pane.  When I switch back to just `net` as the FSAC runtime, I get F# logs again.

**Comment by burak-yildirim** [comment]
I had the same error on Pop_Os (Ubuntu based) with dotnet sdk 6.0 (6.0.300) lately and now somehow it seems fixed. I will tell what i did step by step in case it occurs on someone else's environment.

__Shortly__, i reinstalled for my OS for lower (LTS) version, added ```FSharp.dotnetRoot``` and it worked. 

__Step by step:__ As I said, I am using Pop_Os __21.10__ , which uses Ubuntu 21.10 so my SDK installs were according to that.

- First of all I set ```FSharp.dotnetRoot``` as the error box suggests. __It didn't work__
- So then I deleted the dotnet sdk 6.0 and repeated the installing instructions for my OS since beginning.
  But got error at ```sudo apt install dotnet-sdk-6.0``` part.
- Then I decided to install for older os version __20.04__ (LTS) and deleted ```microsoft-prod.list``` and ```microsoft-prod.list.save``` from ```/etc/apt/sources.list.d``` manually. After successful install, it was still __not enough__.
- Then i set ```FSharp.dotnetRoot``` again, and this time it worked!

dotnet version is still 6.0.300 as before, but I don't have previously installed sdk's commit number so I'm not really sure what changed. I am happy that it works now :)
